src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -2,16 +2,67 @@
 
 import com.chopshop166.chopshoplib.outputs.SendableSpeedController;
 
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.maps.RobotMap;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import java.lang.Math;
+
+/**
+ * What does it do? When the A button is pressed- it shoots a ball. What modes
+ * does it have? Semi-Auto and DUMP. what interactions does it have with other
+ * subsystems? Asks the Indexer and 'vision' if it's ready to shoot. How is it
+ * triggered/OI? A semi-auto button A- or X for shoot all. Does it store any
+ * state? No. Sensors? No.
+ */
 
 public class Shooter extends SubsystemBase {
 
-    private SendableSpeedController shooterWheelMotor;
+    private final SendableSpeedController shooterWheelMotor;
+    public final double shooterHeight;
+    public static double verticalDistance;
+    public final static double GRAVITY = 386.09;
+    public final static double THETA = 37;
+    public final static double TARGET_HEIGHT = 98.25;
+    public static double distanceToTarget;
 
-    public Shooter(RobotMap.ShooterMap map) {
+    public Shooter(final RobotMap.ShooterMap map) {
         super();
+        distanceToTarget = SmartDashboard.getNumber("Distance To Target", 0);
+        shooterHeight = map.shooterHeight();
         shooterWheelMotor = map.shooterWheel();
+        verticalDistance = TARGET_HEIGHT - shooterHeight;
+    }
+
+    public CommandBase spinUp() {
+        return new InstantCommand(() -> {
+            shooterWheelMotor.set(.85); // Make a calculateRPM() function or something to get the speed
+        }, this);
+    }
+
+    public CommandBase spinDown() {
+        return new InstantCommand(() -> {
+            shooterWheelMotor.stopMotor();
```

Comment:
If you're feeling snazzy, you can replace this with `return new InstantCommand(shooterWheelMotor::stopMotor, this);`, it's just a little more terse (but not required)